File: Question 0118 - Pascals Triangle/src/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution
{
    public List<List<Integer>> generate(int numRows)
    {
        List<Integer> row = new ArrayList<>();
        List<List<Integer>> ans = new ArrayList<>();

        for (int i = 0; i < numRows; i++)
        {
            row.add(0,1);
            for (int j = 1; j < row.size()-1; j++)
            {
                row.set(j,row.get(j) + row.get(j+1));
            }

            ans.add(new ArrayList<>(row));
        }

        return ans;
    }
}

File: Question 0118 - Pascals Triangle/src/Solution.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution 
{
public:
    vector<vector<int>> generate(int numRows) 
    {
        vector<vector<int>> ans;
        if(numRows == 0)
            return ans;

        ans.push_back({1});
        for (int i = 1; i < numRows; i++)
        {
            vector<int> curr(i + 1, 1);
            for (int j = 1; j < i; j++)
            {
                curr[j] = ans[i - 1][j - 1] + ans[i - 1][j];
            }

            ans.push_back(curr);
        }

        return ans;
    }
};